base/linalg/bunchkaufman.jl

```diff
@@ -69,11 +69,117 @@ size(B::BunchKaufman, d::Integer) = size(B.LD, d)
+                i += 2
+            else # 'U'
+                p[i - 1], p[-vi] = p[-vi], p[i - 1]
+                i -= 2
+            end
+        end
+    end
+    return p
+end
+
+"""
+    getindex(B::BunchKaufman, d::Symbol)
+
+Extract the factors of the Bunch-Kaufman factorization `B`. The factorization can take the
+two forms `L*D*L'` or `U*D*U'` (or `.'` in the complex symmetric case) where `L` is a
+`UnitLowerTriangular` matrix, `U` is a `UnitUpperTriangular`, and `D` is a block diagonal
+symmetric or Hermitian matrix with 1x1 or 2x2 blocks. The argument `d` can be
+- `:D`: the block diagonal matrix
+- `:L`: the lower triangular factor (if factorization is `L*D*L.'`)
+- `:U`: the lower triangular factor (if factorization is `U*D*U.'`)
+- `:p`: permutation vector
+- `:P`: permutation matrix
+
+```jldoctest
+julia> A = [1 2 3; 2 1 2; 3 2 1]
+3×3 Array{Int64,2}:
+ 1  2  3
+ 2  1  2
+ 3  2  1
+
+julia> F = bkfact(Symmetric(A, :L));
+
+julia> F[:L]*F[:D]*F[:L].' - A[F[:p], F[:p]]
+3×3 Array{Float64,2}:
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+
+julia> F = bkfact(Symmetric(A));
+
+julia> F[:U]*F[:D]*F[:U].' - F[:P]*A*F[:P]'
+3×3 Array{Float64,2}:
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+```
+"""
+function getindex(B::BunchKaufman{T}, d::Symbol) where {T<:BlasFloat}
+    n = size(B, 1)
+    if d == :p
+        return _ipiv2perm_bk(B.ipiv, n, B.uplo)
+    elseif d == :P
+        return eye(T, n)[:,invperm(B[:p])]
+    elseif d == :L || d == :U || d == :D
+        if B.rook
+            # syconvf_rook just added to LAPACK 3.7.0. Uncomment and remove error when we distribute LAPACK 3.7.0
+            # LUD, od = LAPACK.syconvf_rook!(B.uplo, 'C', copy(B.LD), B.ipiv)
+            throw(ArgumentError("reconstruction rook pivoted Bunch-Kaufman factorization not implemented yet"))
+        else
+            LUD, od = LAPACK.syconv!(B.uplo, copy(B.LD), B.ipiv)
+        end
+        if d == :D
+            if B.uplo == 'L'
+                odl = od[1:n - 1]
+                return Tridiagonal(odl, diag(LUD), B.symmetric ? odl : conj.(odl))
+            else # 'U'
+                odu = od[2:n]
+                return Tridiagonal(B.symmetric ? odu : conj.(odu), diag(LUD), odu)
+            end
+        elseif d == :L
+            if B.uplo == 'L'
+                return UnitLowerTriangular(LUD)
+            else
+                throw(ArgumentError("factorization is U*D*U.' but you requested L"))
+            end
+        else # :U
+            if B.uplo == 'U'
+                return UnitUpperTriangular(LUD)
+            else
+                throw(ArgumentError("factorization is L*D*L.' but you requested U"))
+            end
+        end
+    else
+        throw(KeyError(d))
+    end
+end
+
 issuccess(B::BunchKaufman) = B.info == 0
 
 function Base.show(io::IO, mime::MIME{Symbol("text/plain")}, B::BunchKaufman)
     println(io, summary(B))
-    print(io, "successful: $(issuccess(B))")
+    println(io, "D factor:")
```

Comment:
there's an existing one that was showing the raw fields of a BunchKaufman, came up in one of Kristoffer's recent pr's - this will make it fail in a different way